src/main/java/org/springframework/integration/aws/lock/DynamoDbLockRegistry.java

```diff
@@ -0,0 +1,552 @@
+	 * thus should not be shutdown when {@link #destroy()} is called.
+	 */
+	private boolean executorExplicitlySet;
+
+
+	public DynamoDbLockRegistry(AmazonDynamoDB dynamoDB) {
+		this(dynamoDB, DEFAULT_TABLE_NAME);
+	}
+
+	public DynamoDbLockRegistry(AmazonDynamoDB dynamoDB, String tableName) {
+		Assert.notNull(dynamoDB, "'dynamoDB' must not be null");
+		Assert.hasText(tableName, "'tableName' must not be empty");
+
+		this.dynamoDB = dynamoDB;
+		this.tableName = tableName;
+	}
+
+	public DynamoDbLockRegistry(AmazonDynamoDBLockClient dynamoDBLockClient) {
+		Assert.notNull(dynamoDBLockClient, "'dynamoDBLockClient' must not be null");
+
+		this.dynamoDBLockClient = dynamoDBLockClient;
+		this.dynamoDBLockClientExplicitlySet = true;
+		this.dynamoDB = null;
+		this.tableName = null;
+	}
+
+	public void setReadCapacity(long readCapacity) {
+		this.readCapacity = readCapacity;
+	}
+
+	public void setWriteCapacity(long writeCapacity) {
+		this.writeCapacity = writeCapacity;
+	}
+
+	public void setPartitionKey(String partitionKey) {
+		Assert.hasText(partitionKey, "'partitionKey' must not be empty");
+		this.partitionKey = partitionKey;
+	}
+
+	/**
+	 * Specify a name of the table attribute which is used as a sort key.
+	 * @param sortKeyName the sort key attribute name to use.
+	 */
+	public void setSortKeyName(String sortKeyName) {
+		this.sortKeyName = sortKeyName;
+	}
+
+	/**
+	 * Specify a value for the sort key attribute of the lock item.
+	 * @param sortKey the sort key value to use.
+	 */
+	public void setSortKey(String sortKey) {
+		this.sortKey = sortKey;
+	}
+
+	public void setLeaseDuration(long leaseDuration) {
+		this.leaseDuration = leaseDuration;
+	}
+
+	public void setHeartbeatPeriod(long heartbeatPeriod) {
+		this.heartbeatPeriod = heartbeatPeriod;
+	}
+
+	public void setRefreshPeriod(long refreshPeriod) {
+		this.refreshPeriod = refreshPeriod;
+	}
+
+	/**
+	 * Set the {@link Executor}, where is not provided then a default of
+	 * cached thread pool Executor will be used.
+	 * @param executor the executor service
+	 */
+	public void setExecutor(Executor executor) {
+		this.executor = executor;
+		this.executorExplicitlySet = true;
+	}
+
+	@Override
+	public void afterPropertiesSet() {
+		if (!this.dynamoDBLockClientExplicitlySet) {
+			AmazonDynamoDBLockClientOptions dynamoDBLockClientOptions =
+					AmazonDynamoDBLockClientOptions
+							.builder(this.dynamoDB, this.tableName)
+							.withPartitionKeyName(this.partitionKey)
+							.withSortKeyName(this.sortKeyName)
+							.withHeartbeatPeriod(this.heartbeatPeriod)
+							.withLeaseDuration(this.leaseDuration)
+							.build();
+
+			this.dynamoDBLockClient = new AmazonDynamoDBLockClient(dynamoDBLockClientOptions);
+		}
+
+		this.leaseDuration =
+				(long) new DirectFieldAccessor(this.dynamoDBLockClient)
+						.getPropertyValue("leaseDurationInMilliseconds");
+	}
+
+	@Override
+	public boolean isAutoStartup() {
+		return true;
```

Comment:
Why not configurable?

Reply:
If you take a look into the `start()`, there is nothing just table creation on the AWS: I just moved the hard network job into the `start()` instead of `afterPropertiesSet()`. 
Therefore I don't see reason to have this not started automatically - it isn't going to be useful without created table. Or... the table must be provided in advance.

I can't reconsider this if there is really a strong feeling on the matter.

By the way: is it going to work just with the `Lifecycle`? I mean this will be as a bean and container will start it anyway. The point is do not expose lifecycle control to end user at all.

Thanks

Reply:
Another inconvenience that this `AmazonDynamoDBLockClient` is not async and we really can't initiate request and exit, like we do in the `DynamoDbMetaDataStore.afterPropertiesSet()`.

Reply:
OK; thanks for the explanation; no `Lifecycle` beans are not started unless the user `start()`s the container.

Reply:
OK! I decided to remove lifecycle control in favor of the separate thread execution in the `afterPropertiesSet()`.
See an update in commits.